### PR TITLE
Orca: move unit test `write_read_imagenet`

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change

This PR fixes the following issue:

+ Test function `test_write_read_imagenet` in [`test_spark_backend.py`](https://github.com/intel-analytics/BigDL/blob/main/python/orca/test/bigdl/orca/data/test_spark_backend.py#L115) requires tf version <= 1.15, which is not compatible with python 3.8
    + dependency files: https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/data/image/imagenet_dataset.py
    + Previous PR: https://github.com/intel-analytics/BigDL/pull/7379

+ Current python 3.8 unit test ignores the script `test_spark_backend.py`, while other test functions in the script should not be ignored.

### 2. Summary of the change

+ create a new test sciprt `test/bigdl/orca/tfpark/test_tfrecord.py`, which will be tested in `Orca-Python-Py38/Py37-Spark3`
+ move `test_write_read_imagenet()` from `data/test_spark_backend.py` to `tfpark/test_tfrecord.py`
+ re-enable py38 test for `test_spark_backend.py`

### 3. How to test

unit test
- [x] [Orca-Python-Py38-Spark3](https://github.com/intel-analytics/BigDL/actions/runs/4133377045/jobs/7143744196)
- [x] [Orca-Python-Py37-Spark3](https://github.com/intel-analytics/BigDL/actions/runs/4133377045/jobs/7143743872)
- [x] [Orca-Python-Ray-Py38-Spark3](https://github.com/intel-analytics/BigDL/actions/runs/4133377045/jobs/7143744375)
- [x] [Orca-Python-Ray-Py37-Spark3](https://github.com/intel-analytics/BigDL/actions/runs/4133377045/jobs/7143744019)

### CODE REVIEW

**Comment by hkvision** [comment]
Then you need to move this test to a place together with tf1.15 tests and recover test_spark_backend for python3.8.

File: python/orca/test/bigdl/orca/data/test_spark_backend.py [hunk 1]
```diff
@@ -112,16 +112,6 @@ def test_read_parquet(self):
 
         shutil.rmtree(temp)
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/data/test_spark_backend.py:114]: remove one empty line

Comment (hamham223) [reply python/orca/test/bigdl/orca/data/test_spark_backend.py:114]: removed, see the latest commit: https://github.com/intel-analytics/BigDL/blob/05ce85d99e6652873da62bf3152e8a9d9a4c11b4/python/orca/test/bigdl/orca/data/test_spark_backend.py

File: python/orca/test/bigdl/orca/tfpark/test_tfrecord.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from unittest import TestCase
+import shutil
+
+from bigdl.dllib.nncontext import *
+from bigdl.orca.data.image import write_tfrecord, read_tfrecord
+from bigdl.orca.data.utils import *
+from bigdl.orca.data.transformer import *
+
+
+class TestXShardsSparkBackend(TestCase):
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/tfpark/test_tfrecord.py:29]: TestTFRecord

File: python/orca/test/bigdl/orca/tfpark/test_tfrecord.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import tempfile
+import os.path
+import pytest
+from unittest import TestCase
+import shutil
+
+from bigdl.dllib.nncontext import *
+from bigdl.orca.data.image import write_tfrecord, read_tfrecord
+from bigdl.orca.data.utils import *
+from bigdl.orca.data.transformer import *
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/tfpark/test_tfrecord.py:26]: don't need these imports?

File: python/orca/test/bigdl/orca/tfpark/test_tfrecord.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# limitations under the License.
+#
+
+import tempfile
+import os.path
+import pytest
+from unittest import TestCase
+import shutil
+
+from bigdl.dllib.nncontext import *
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/tfpark/test_tfrecord.py:23]: need this import?

**Pull request merged**